### PR TITLE
add search functionality of pokemon

### PR DESCRIPTION
**Pull request by rmoise**
This feature will add a search bar on the website and allow the user to filter through pokemon.

### CODE REVIEW

File: js/scripts.js [hunk 1]
```diff
@@ -122,6 +136,46 @@ let pokemonRepository = (function () {
...
+        let filterUpperCase = searchBar.value.toUpperCase();
+
+        pokemonList.forEach(function (pokemon) {
+            if (pokemon.innerText.toUpperCase().indexOf(filterUpperCase) > -1) {
+                pokemon.style.display = '';
+            } else {
+                pokemon.style.display = 'none';
+            }
+        });
+    });
```
Comment (oxenprogrammer) [inline js/scripts.js]: This is nicely done. How about we display `No Pokemon Found from Your Search` when the search is empty? WDYT

Comment (rmoise) [reply js/scripts.js]: Thank you! Good idea, just now added the revised version.

File: js/scripts.js [hunk 2]
```diff
@@ -141,12 +141,21 @@ let pokemonRepository = (function () {
...
+                showError = false;
             } else {
+                // makes element invisible and hides error message
                 pokemon.style.display = 'none';
+                $('.error').hide();
+            }
+            if (showError) {
+                $('.error').show();
+            } else {
+                $('.error').hide();
```
Comment (oxenprogrammer) [inline js/scripts.js]: `LGTM` We can also use a **ternary operator** like this `showError ? $('.error').show() : $('.error').hide();` since it is short.

**Review by oxenprogrammer — APPROVED** [review]

File: js/scripts.js (hunk 2 above)
Comment (rmoise) [reply js/scripts.js]: awesome, thanks! just made the change.

**Pull request merged**